src/status_im2/contexts/chat/bottom_sheet_composer/keyboard.cljs

```diff
@@ -0,0 +1,74 @@
+            [react-native.async-storage :as async-storage]
+            [react-native.core :as rn]
+            [react-native.platform :as platform]
+            [react-native.reanimated :as reanimated]))
+
+(defn get-kb-height
+  [curr-height default-height]
+  (if (and default-height (< curr-height default-height))
+    default-height
+    curr-height))
+
+(defn store-kb-height
+  [{:keys [kb-default-height]} keyboard-height]
+  (when (and (not @kb-default-height) (pos? keyboard-height))
+    (async-storage/set-item :kb-default-height keyboard-height)))
+
+(defn handle-emoji-kb-ios
+  [e
+   {:keys [emoji-kb-extra-height]}
+   {:keys [text-value]}
+   {:keys [height saved-height]}
+   {:keys [max-height]}]
+  (let [start-h   (oops/oget e "startCoordinates.height")
+        end-h     (oops/oget e "endCoordinates.height")
+        diff      (- end-h start-h)
+        max       (- max-height diff)
+        curr-text @text-value]
+    (if (> (reanimated/get-shared-value height) max)
+      (do
+        (reanimated/set-shared-value height (- (reanimated/get-shared-value height) diff))
+        (reanimated/set-shared-value saved-height (- (reanimated/get-shared-value saved-height) diff))
+        (reset! text-value (str @text-value " "))
+        (js/setTimeout #(reset! text-value curr-text) 0)
+        (reset! emoji-kb-extra-height diff))
+      (when @emoji-kb-extra-height
+        (reanimated/set-shared-value height
+                                     (+ (reanimated/get-shared-value height) @emoji-kb-extra-height))
+        (reanimated/set-shared-value saved-height
+                                     (+ (reanimated/get-shared-value saved-height)
+                                        @emoji-kb-extra-height))
+        (reset! emoji-kb-extra-height nil)))))
+
+(defn add-kb-listeners
+  [{:keys [keyboard-show-listener keyboard-frame-listener keyboard-hide-listener input-ref] :as props}
+   state animations dimensions keyboard-height]
+  (reset! keyboard-show-listener (.addListener rn/keyboard
+                                               "keyboardDidShow"
+                                               #(store-kb-height state keyboard-height)))
+  (reset! keyboard-frame-listener (.addListener
+                                   rn/keyboard
+                                   "keyboardWillChangeFrame"
+                                   #(handle-emoji-kb-ios % props state animations dimensions)))
+  (reset! keyboard-hide-listener (.addListener rn/keyboard
+                                               "keyboardDidHide"
+                                               #(when platform/android?
+                                                  (.blur ^js @input-ref)))))
+
+(defn handle-refocus-emoji-kb-ios
+  [{:keys [saved-emoji-kb-extra-height]}
+   {:keys [height saved-height last-height]}
+   {:keys [lines max-lines]}]
+  (when @saved-emoji-kb-extra-height
+    (js/setTimeout (fn []
+                     (when (> lines max-lines)
```

Comment:
I am not sure what is the point of the second check? If it was `nil` we will be setting it to `nil` anyways

Reply:
I agree with @ilmotta but I am more of the approach of cancelling timers instead of adding extra checks, we could store the timer reference and make usage of `js/clearTimeout` if timeout is not needed anymore or a new one is set.

Reply:
> I am not sure what is the point of the second check? If it was nil we will be setting it to nil anyways

Right, in this example it's really not that important, since adding nil to a number in Clojure does nothing, it's like adding a zero to the sum. But the point in other situations is that the timer will run some time after the timer is created, which means in the meantime, other parts of the code can make the atom value nil. It's quite easy to replicate this behavior, it's just a matter of the ref'ed componented be unmounted before the timer runs (but after it was scheduled).

I agree with @briansztamfater, as I also made a similar comment, it's better to clear the timeout and avoid this problem altogether. But still, the point I mentioned above is important to take in consideration in cases where atoms can be nil and some of them may survive between mount/unmount cycles.

Reply:
Alright, I will consider clearing the timer.

Btw, this method is used to handle an edge case (if the user collapses a maximized input while on emoji keyboard on iOS). So refocusing again in less than 600ms after executing this method is an edge case of an edge case :)